### DISCUSSION TITLE
How can binding event by condition?

### DISCUSSION THREAD

**Discussion by dbwodlf3** [lit/lit#3134] [Q&A] [answered]
[releated issue](https://github.com/lit/lit/issues/625)

How can i binding event by condition?
I can solve this is just make two html, but i don't like to do that.

**Comment by augustjk** [comment]
~~I think the `ifDefined()` mentioned in the issue you linked also works for event bindings.~~
~~https://lit.dev/playground/#gist=5b1b1ce18b040a4c9e3e4753f1eecc56~~

Edit: See @sorvell 's answer below.

**Comment by sorvell** [comment, marked as answer]
You can do this easily by putting the condition in the handler:

`<button @click=${this.shouldHandleClick && this.handleClick}>...`

See: https://lit.dev/playground/#gist=e1b79a1e5d0d492445cee0288f1fca4c

**Reply by augustjk** [reply]
Nice! I guess `ifDefined()` isn't even necessary for event binding since it's not an attribute that needs to be removed.

**Reply by justinfagnani** [reply]
You can also just put the condition in the handler:

```ts
private _handleClick(e) {
  if (!this.shouldHandleClick) {
    return;
  }
}
```

**Reply by ynxtynxt** [reply]
BTW I tried having 2 `@click` for no good reason tho the result was a little unexpected.
`<div @click=... @click=...>` not sure if it is worth noting.
